clang/lib/AST/Interp/ByteCodeExprGen.h

```diff
@@ -129,7 +129,13 @@ class ByteCodeExprGen : public ConstStmtVisitor<ByteCodeExprGen<Emitter>, bool>,
 
   /// Classifies a type.
   std::optional<PrimType> classify(const Expr *E) const {
-    return E->isGLValue() ? PT_Ptr : classify(E->getType());
+    if (E->isGLValue()) {
+      if (E->getType()->isFunctionType())
+        return PT_FnPtr;
+      return PT_Ptr;
+    }
+
+    return classify(E->getType());
```

Comment:
This feels unrelated to the patch title; can you explain it a bit more? (It doesn't seem wrong, just not seeing how it ties to the test behavior changes.)

Reply:
This is basically just an oversight;  This should've been added when we added function pointers. This case was just never exercised. I can tell you what test case trips this up as soon as my local build finished.

Reply:
Ah, let's pull those changes from this PR and land them separately then.